include/galois.h

```diff
@@ -46,57 +46,69 @@
 extern "C" {
 #endif
 
-extern int galois_init_default_field(int w);
-extern int galois_uninit_field(int w);
-extern void galois_change_technique(gf_t *gf, int w);
+typedef struct gf2_t_ {
+	gf_t* gf;
```

Comment:
I don't get why this member is necessary.
I think it's okay to remove this member and make `galois_init_field` return both `gf2_t` and `gf_t` created.
I'm commenting because it seems that `gf2_t` is the global state packed in a struct, and `gf_t* gf` created in `galois_init_field` doesn't look like a global state and not used elsewhere in this repository.